### PR TITLE
fix: Include ConnectionError and urllib3 exception as retriable

### PR DESCRIPTION
**Pull request by andrewsg**

### CODE REVIEW

File: google/resumable_media/_helpers.py [hunk 1]
```diff
@@ -384,10 +384,13 @@ def _get_connection_error_classes():
...
 
     import requests.exceptions
+    import urllib3.exceptions
 
     return (
         requests.exceptions.ConnectionError,
         requests.exceptions.ChunkedEncodingError,
+        urllib3.exceptions.ProtocolError,
+        ConnectionError,  # Python 3.x only, superclass of ConnectionResetError.
     )
```
Comment (tseaver) [inline google/resumable_media/_helpers.py:394]: Given that we are adding a stdlib type here, it seems like perhaps having this method raise `ImportError` is no longer "correct":  instead, perhaps the imports themselves should be conditional.

File: tests/unit/test__helpers.py [hunk 2]
```diff
@@ -20,6 +20,7 @@
 import mock
 import pytest
 import requests.exceptions
+import urllib3.exceptions
```
Comment (tseaver) [inline tests/unit/test__helpers.py:23]: Having these imports outside of the `requests` subpackage is a bit manky, too.

**Review by tseaver — COMMENTED** [review]
Rather than the changes I suggested inline, perhaps it makes more sense to move `wait_and_retry` and the associated helpers down into the `requests` sub-package?

**Comment by andrewsg** [comment]
@tseaver I decided I agree with you on the architecture question; I don't remember what I was thinking when I put it in _helpers originally, but I've refactored retries to be in _request_helpers instead. I think that will resolve your feedback. PTAL

**Review by tseaver — APPROVED** [review]

**Pull request merged**